typechecker: Added cint type id to negate

Lenart12 opened the pull request:
Before we couldn't typecast negative constants to c_int.
This fix makes it so now we can do things like `-1 as! c_int`.
Fixes #465.

linusg commented on tests/typechecker/array_negative_c_int.jakt:
```diff
@@ -0,0 +1,10 @@
+/// - output: "-1\n-1\n2\n"
+
+function main() {
+    let array: [c_int] = [-1 as! c_int, -1 as! c_int, 2 as! c_int]
```
What's the point of doing the same test twice here?

linusg reviewed (changes requested):
https://github.com/SerenityOS/jakt/blob/main/CONTRIBUTING.md#code-submission-policy
> Write the commit message subject line in the imperative mood ("Foo: Change the way dates work", not "Foo: Changed the way dates work").

Lenart12 commented:
Changed the commit message and changed up the test.
`if array[0] == -1 as! c_int` still feels weird, but atleast its castable now.
Would be cool if we could do just `if array[0] == -1` since casting -1 to c_int would never fail.

linusg commented on tests/typechecker/array_negative_c_int.jakt:
```diff
@@ -0,0 +1,11 @@
+function main() {
+    let array: [c_int] = [-1 as! c_int, 2 as! c_int]
+
+    if array[0] == -1 as! c_int {
```
I don't understand why this is needed. Just print `array[0]` unconditionally...?

linusg reviewed (changes requested):
`Adds` is not imperative mood

linusg reviewed (approved):
Thanks!
